src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+        # If you repository contains multiple packages consider a pattern like:
+        # - 'my_package_name-v[0-9]+.[0-9]+.[0-9]+*'
+
+    # Publish using the reusable workflow from dart-lang.
+    jobs:
+      publish:
+        uses: dart-lang/setup-dart/.github/workflows/publish.yml
+        # with:
+        #   working-directory: path/to/package/within/repository
+    ```
+
+Make sure to match the pattern in `on.push.tags` with the `tag-pattern`
+specified on pub.dev. Otherwise, the GitHub Action workflow won’t work.
+
+<!--
+TODO: The file described in the following passage does not exist.
+We can re-include it once the file has been created and merged.
+
+The preceding workflow file uses
+`dart-lang/setup-dart/.github/workflows/publish.yml` to publish the package.
+This _[reusable workflow][3]_ allows the Dart team to maintain
+the publishing logic and enables `pub.dev` to know how the package was published.
+Though optional, the Dart team encourages using this reusable workflow.
+-->
+
+### Create a custom GitHub Action workflow to publish to pub.dev
+
+If you need generated code in your package, check this generated code
+into your repository. This simplifies verifying if the files
+published on `pub.dev` match the files from your repository. If you cannot
+check in generated or built artifacts into your repository, you
+can create a custom workflow. This workflow could resemble the
+following YAML markup:
+
+```yaml
+# .github/workflows/publish.yml
+name: Publish to pub.dev
+
+on:
+  push:
+    tags:
+    - 'v[0-9]+.[0-9]+.[0-9]+*' # tag pattern on pub.dev: 'v{{version}}'
+
+# Publish using custom workflow
+jobs:
+  publish:
+    permissions:
+      id-token: write # This is required for authentication using OIDC
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dart-lang/setup-dart
+      - name: Install dependencies
+        run: dart pub get
+      # Here you can insert custom steps you need
+      # - run: dart tool/generate-code.dart
+      - name: Publish
+        run: dart pub publish --force
+```
+
+The workflow authenticates to `pub.dev` using a temporary [token][1].
+Create and configure this token in the `dart-lang/setup-dart` `step` in
+the `publish.yaml` file.
+To publish to `pub.dev`, subsequent steps can run
+`dart pub publish --force`.
+
+{{site.alert.note}}
+  At this point in this guide, anyone with push access to your
+  repository can publish new versions of the package.
+  Consider using [tag protection rules][sec-gh-tag-protection] or
+  [GitHub deployment Environments][sec-gh-environment] to limit who can
+  publish.
+{{site.alert.end}}
+
+[sec-gh-tag-protection]: #hardening-security-with-tag-protection-rules-on-github
+[sec-gh-environment]: #hardening-security-with-github-deployment-environments
+
+
+## Trigger automated publishing using Github Actions
+
+After you have completed the preceding activities, you can publish a
```

Comment:
Is it really unreasonable to say:
> After you've configured automated publishing on `pub.dev` and created a
> GitHub Actions workflow

instead of  "the preceding activities"

I see how "the preceding activities" is shorter, but also not entirely clear what we're referring.
If we had support for proper section references I would write "After completeting (1) and (2)".
Otherwise, it's kind of hard to guess what we're talking about, and I'm not sure we want to leave users guessing.